### PR TITLE
Fix load and set of previous language and scroll

### PR DESCRIPTION
**Pull request by joaoeudes7**
This PR resolve the issue #12, open for any change or improvement!

### CODE REVIEW

**Comment by keiji** [comment]
Hi, Thank you for your pull request!
I will review the PR over the weekend.

File: app/src/main/java/jp/co/c_lis/ccl/morelocale/ui/locale_list/LocaleListViewModel.kt [hunk 1]
```diff
@@ -60,6 +64,8 @@ class LocaleListViewModel @Inject constructor(
 
     fun setLocale(localeItem: LocaleItem) {
         viewModelScope.launch {
+            MoreLocale.setLocale(localeItem.locale)
+
             preferenceRepository.saveLocale(localeItem)
         }
```
Comment (keiji) [inline app/src/main/java/jp/co/c_lis/ccl/morelocale/ui/locale_list/LocaleListViewModel.kt:70]: If the app doesn't have permission. `MoreLocal.setLocale` method throws `InvocationTargetException`.
The caller cannot catch the exception outside CoroutineScope.

```
5279-5279/jp.co.c_lis.ccl.morelocale E/AndroidRuntime: FATAL EXCEPTION: main
    Process: jp.co.c_lis.ccl.morelocale, PID: 5279
    java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Native Method)
        at jp.co.c_lis.morelocale.MoreLocale.setLocaleList(MoreLocale.java:118)
        at jp.co.c_lis.morelocale.MoreLocale.setLocale(MoreLocale.java:87)
        at jp.co.c_lis.ccl.morelocale.ui.locale_list.LocaleListViewModel$setLocale$1.invokeSuspend(LocaleListViewModel.kt:67)
```

I think it is a good idea to put `MoreLocale.setLocale` inside the ViewModel!

And could you consider that call `MoreLocale.setLocale` from outside `viewModelScope`. Or to respond even more better, I think there is a way to add `MutableLiveData` as  `eventPermissionRequired` to `LocaleListViewModel`.

```suggestion
        MoreLocale.setLocale(localeItem.locale)

        viewModelScope.launch {
            preferenceRepository.saveLocale(localeItem)
        }
```

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -16,17 +17,25 @@
...
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
 
-        <receiver android:name=".receiver.BootCompletedReceiver">
+        <receiver
+            android:name=".receiver.BootCompletedReceiver"
+            android:enabled="true"
+            android:exported="true">
```
Comment (keiji) [inline app/src/main/AndroidManifest.xml:33]: 👍

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -16,17 +17,25 @@
...
             <intent-filter>
                 <action android:name="android.intent.action.BOOT_COMPLETED" />
+                <action android:name="android.intent.action.QUICKBOOT_POWERON" />
+                <action android:name="android.intent.action.REBOOT"/>
```
Comment (keiji) [inline app/src/main/AndroidManifest.xml:37]: `BootCompletedReceiver` filtered intent by `BOOT_COMPLETED`. Should the filter be removed?

https://github.com/c-lis/morelocale/blob/8ecbc19a69bf501f540fdfcade6463018f1b2420/app/src/main/java/jp/co/c_lis/ccl/morelocale/receiver/BootCompletedReceiver.kt#L15-L23

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
     <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
     <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
+    <uses-permission android:name="android.permission.WAKE_LOCK" />
```
Comment (keiji) [inline app/src/main/AndroidManifest.xml:7]: Hmm... I don't understand what this permission for?

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (joaoeudes7) [reply app/src/main/AndroidManifest.xml:37]: Yes, you're right!

Comment (joaoeudes7) [reply app/src/main/AndroidManifest.xml:37]: But direct-boot may be interesting? https://developer.android.com/training/articles/direct-boot

File: app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (joaoeudes7) [reply app/src/main/AndroidManifest.xml:7]: Sorry, auto suggestion from IDE, removed!

**Comment by emad54** [comment]
لطفت می خواهم زبان موبایل من به فارسی تبدیل شود از شما سپاس گذارم

**Comment by emad54** [comment]
من دوست دارم زبان موبایل به زبان خودم فارسی ایران باشد از شما تشکر می کنم

**Review by keiji — APPROVED** [review]

**Pull request merged**